src/React/Basic.purs

```diff
@@ -369,6 +373,46 @@ foreign import toReactComponent
   -> { render :: Self props state -> JSX | spec }
   -> ReactComponent { | jsProps }
 
+type ReactContext a =
+  { provider :: ReactComponent { value :: a, children :: Array JSX }
+  , consumer :: ReactComponent { children :: a -> Array JSX }
+  }
+
+-- | Create a `ReactContext` given a default value. Use `provider` and `consumer`
+-- | to provide and consume context values. Alternatively, use the fields of
+-- | `ReactContext` directly if a `ReactComponent` is required for interop.
+-- |
+-- | ```purs
+-- | render self =
+-- |   R.div_
+-- |   [ R.button
+-- |     { onClick: capture_ $ self.setState \s -> s { counter = s.counter + 1 }
+-- |     , children: [ R.text "Tick!" ]
+-- |     }
+-- |   , provider countContext self.state.counter
+-- |     [ consumer countContext \counter ->
+-- |        [ R.text $ "Ticks: " <> (show counter)
+-- |        ]
+-- |     ]
+-- |   ]
+-- | ```
+-- |
+-- | __*See also:* `provider`, `consumer`, React's documentation regarding Context__
+foreign import createContext :: forall a. a -> ReactContext a
+
+-- | Create a provider `JSX` given a context value and children.
+-- |
+-- | __*See also:* `createContext`, `consumer`__
+provider :: forall a. ReactContext a -> a -> Array JSX -> JSX
+provider context value children =
+  element context.provider { value, children }
+
+-- | Create a consumer `JSX` from a context value to children.
+-- |
+-- | __*See also:* `createContext`, `producer`__
+consumer :: forall a. ReactContext a -> (a -> Array JSX) -> JSX
```

Comment:
Both `provider` and `consumer` use `Array JSX`. Should they just be `JSX`? (I was under the impression only a single child was allowed)

Reply:
Nevermind, just checked this and it works